crates/codegen/runtime/cargo/crate/src/runtime/compilation/mod.rs

```diff
@@ -1,7 +1,15 @@
+//! The `compilation` module provides some basic utilities that users can use to combine multiple source files
+//! into a single compilation unit. The compilation unit can then be used to get a complete [`BindingGraph`][`crate::bindings::BindingGraph`]
+//! that can resolve references and defintions across all the added files.
+//!
+//! This API is a work in progress. For now, it's recommended to use our [Typescript API](https://nomicfoundation.github.io/slang/latest/user-guide/07-semantic-analysis/01-compilation-units/#71-compilation-units)
+//! to build a [`CompilationUnit`] and resolve binding graphs.
```

Comment:
What's the plan for the `CompilationUnit`? As I said in my last review, it's unusable from Rust unless the user enables the `__private_wasm_apis` feature flag, because it's not possible to build one. If we're going to resolve that issue in a future PR, I'd suggest we put `CompilationUnit` and related files under the same feature flag for consistency. Otherwise it will be very confusing to users.

Reply:
Is there a link to that comment? I'm not sure why we need `__private_wasm_apis` here.